In [ ]:
from vampyr import vampyr3d as vp
import numpy as np
import matplotlib.pyplot as plt
# from copy import deepcopy

# import KAIN
import utils
# from spinor import spinor
from scfsolv import scfsolv as ss
from scfsolv import scfsolv_1stpert as ss1

prec = 1.0e-4
khist = 4
thrs = prec
c =  137.035999177
# c = 60

N_components = 2
# Pord = 1 #maximum pertubation order computed

# r_x = np.linspace(-16, 16, 10000) # create an evenly spaced set of points between -0.99 and 0.99
# r_y = np.linspace(-16, 16, 10000)
# r_z = np.linspace(-16, 16, 10000)
# r = [r_x, r_y, r_z]
# Restricted = True


# Electric_field = [0.01, 0., 0.]
Electric_field = [0.01, 0., 0.]
# Analytic nuclear potential
# print("pouet")
#H
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [1], "/home/qpitto/Tests_KAIN/test_H/norun/orbitals/", 1, 1, False #Initial guess 
#He 
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [2], "/home/qpitto/Tests_KAIN/test_He/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [2], "/home/qpitto/Tests_KAIN/test_He/run/orbitals/", 2, 2, False #Converged NR result
#H2
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.73699313, 0.100, 0.100], [-0.73699313, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1., 0.100, 0.100], [-1., 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.7, 0.100, 0.100], [-0.7, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.6992, 0.100, 0.100], [-0.6992, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 2, 1, False #Initial guess
#H2+ 
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1., 0.100, 0.100], [-1., 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 1, 1, False #Initial guess
# H2plus_dirac = -1.1026415810325771641
# lambda_factor = (2*c**2 + H2plus_dirac)/(2*c**2)
# R, Z, pathMol, nElectron, InitGuessType, scalar = [[1/lambda_factor, 0.100, 0.100], [-1/lambda_factor, 0.100, 0.100]], [1,1], "/home/qpitto/Tests_KAIN/test_H2/norun/orbitals/", 1, 1, False #Initial guess
#LiH
# R, Z, pathMol, nElectron = [[1.508001,  0.188973, 0.188973], [-1.508001,  0.188973, 0.188973]], [3,1], "/home/qpitto/Tests_KAIN/test_LiH/norun/orbitals/", 2 #Initial guess
# R, Z, pathMol, nElectron = [[1.508001,  0.188973, 0.188973], [-1.508001,  0.188973, 0.188973]], [3,1], "/home/qpitto/Tests_KAIN/test_LiH/run/orbitals/", 2 #LiH
#Ne
# R, Z, pathMol, nElectron = [[0.100, 0.100, 0.100]], [10], "/home/qpitto/Tests_KAIN/test_Ne/norun/orbitals/phi_p_scf_", 5 #Ne
R, Z, pathMol, nElectron, InitGuessType, scalar = [[0.1, 0.100, 0.100]], [10], "/home/qpitto/Tests_KAIN/test_He/norun/orbitals/", 2, 1, False
#H2O
# R, Z, pathMol, nElectron = [[0.0000,  0.000, -0.125], [-1.4375,  0.000,  1.025], [1.4375,  0.000,  1.025]], [8,1,1], "/home/qpitto/Tests_KAIN/test_water/run/orbitals/", 5 #H2O
# R, Z, pathMol, nElectron = [[0.0000,  0.000, -0.125], [-1.4375,  0.000,  1.025], [1.4375,  0.000,  1.025]], [8,1,1], "/home/qpitto/Tests_KAIN/initial_guesses/phi_p_scf_", 5 #H2O
#CH4
# R, Z, pathMol, nElectron = [[0.1, 0.1, 0.1], [0.1, 0.1, 2.15413215], [2.03631019, 0.1, -0.5856834], [-0.54635362, 1.92524633, -0.5856834], [-0.54635362, -1.72524633, -0.5856834]], [6,1,1,1,1], "/home/qpitto/Tests_KAIN/test_CH4/norun/orbitals/", 5 #CH4

nz = len(Z)

#create scf solver instance with default MRA
Scfs = ss(N_components, prec, khist, scalar=scalar, lightspeed=c)

# print("testLC: ", utils.LeviCivita([1,2,3]), utils.LeviCivita([1,1,3]), utils.LeviCivita([2,1,3]))
# from spinor import spinor
# print("Test spinors")
# spineur = spinor(Scfs.mra, 2)
# spineur.setZero()
# print("length", len(spineur))
# spindeur = spineur + spineur
# print("+ ok")
# spindeur = spineur - spineur
# print("- ok")
# spindeur = 1. * spineur
# spindeur = 2 * spineur
# spindeur = 1j * spineur
# spindeur = spineur * 1
# print("* ok")
# spindeur = spineur / 2
# spindeur = spineur / 2.
# print("norme = ", spineur.compSqNorm())
# print("dot = ", spineur.dot(spindeur))
# spindeur = spineur.derivative(0)
# print("Derivative ok")


# tut = [[-1.514986e+00, -8.879809e-02],
# [-8.879809e-02, 1.188349e+00]]
# print(np.linalg.eig(tut))

#initialise molecule geometry and properties
print("start initMolec")
Scfs.initMolec(nElectron, R, Z, pathMol, InitGuessType, scalar)
print("init done")
#run the SCF solver
Scfs.scfRun(thrs, True, True)

# Scfs_2C = ss(N_components, 0.01*prec, khist, scalar = False)
# reprojected_copy = Scfs.phi_prev[0][-1].reproject(Scfs_2C.P_eps)
# # print("tutbim")
# # Scfs_2C.initMolec(nElectron, R, Z, Scfs, restricted=False, copy_orbitals=True)
# # Scfs_2C.scfRun(thrs, True, True)

# print("Sanity test")
# for i in range(len(Scfs.phi_prev)):
#     for j in range(len(Scfs.phi_prev)):
#         print(f"Test: {i}, {j}", vp.dot(Scfs.phi_prev[i][-1], Scfs.phi_prev[j][-1]))

# print("Start perturbation")
# Scfs_pert = ss1(Scfs)

# print("Contructor passed")
# Scfs_pert.initMolec(Electric_field)
# # Scfs_pert.print_operators()

# Scfs_pert.scfRun(thrs, True, True)

start initMolec
Initmolec
S initial guess =  [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (50.359150303321584+0j)
Expectation value gradk*Nab :  (-0.2459675400625521+0j)
Expectation value sporb :  6.029769739012778e-19j
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (50.359150303321584+0j)
Expectation value gradk*Nab :  (-0.2459675400625521+0j)
Expectation value sporb :  6.029769739012778e-19j
Fock [[-43.86346465-6.0297988e-19j]]
Kinetic [[49.88643466-6.02981707e-19j]]
Vnuc [[-99.99988484+0.j]]
Coulomb [[12.49997105+0.j]]
Exchange [[6.24998553+0.j]]


/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:170: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:650: ComplexWarning: Casting complex values to real discards the imaginary part
  VzFphi = V_z/(self.c**2) * self.Fock[orb, orb] * self.phi_prev[orb][-1]


Power Iter, norm of new guess =  (1.0105060606247223+0j) (1.0105060606247223+0j) 0j
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-43.86346465-6.0297988e-19j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (48.63738981843436+0j)
Expectation value gradk*Nab :  (-0.23759913823750117+0j)
Expectation value sporb :  1.1286464686540973e-18j


/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:505: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)


Power Iter, norm of new guess =  (1.0002811739627087+0j) (1.0002811739627087+0j) 0j
orthonorm
S durant Ortho [[1.00028117+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (47.29448342689536+0j)
Expectation value gradk*Nab :  (-0.22917876066974566+0j)
Expectation value sporb :  4.8384628285955865e-18j
Fock [[-44.01967784-4.83845624e-18j]]
Kinetic [[46.85604922-4.83842918e-18j]]
Vnuc [[-96.90765543+0.j]]
Coulomb [[12.06385675+0.j]]
Exchange [[6.03192837+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (47.29448342689536+0j)
Expectation value gradk*Nab :  (-0.22917876066974566+0j)
Expectation value sporb :  4.8384628285955865e-18j
Fock [[-44.01967784-4.83845624e-18j]]
Kinetic [[46.85604922-4.83842918e-18j]]
Vnuc [[-96.90765543+0.j]]
Coulomb [[12.06385675+0.j]]
Exchange [[6.03192837+0.j]]
Orbital: 0    Norm: [0.99621206]    Update: [0.01444899]    Energy:[-43.96634375-1.12861768e-18j]
Scaled energy (4C Dirac energy) [-43.91493545]
====

In [ ]:
# Scfs_2C = ss(N_components, 0.01*prec, khist, scalar = False)
# # reprojected_copy = Scfs.phi_prev[0][-1].reproject(Scfs_2C.P_eps)
# print("tutbim")
# Scfs_2C.initMolec(nElectron, R, Z, Scfs, restricted=False, copy_orbitals=True)
# print("===============Init done=============================================")
# Scfs_2C.scfRun(thrs, True, True)

In [ ]:
import numpy as np
# F= [[-2.7896489 -2.66352263e-07j, 0.00707083+5.25880281e-08j],[ 0.00704288-1.01844084e-07j, -1.11828297+1.44609334e-08j]] #initial
F=np.array([[-2.7263095 -9.37847108e-05j, -0.04998474+1.23454107e-03j],
 [-0.05001107-9.24036599e-04j, -1.19366291-3.31073015e-06j]]) #final (no kain)
eigvals, U = np.linalg.eig(F)
print(eigvals)

[-2.72793954-8.35624316e-05j -1.19203287-1.35330094e-05j]


In [ ]:
# # print("Scaled energy (4C Dirac energy)", (1/(-0.50056707) - 1/(c**2))**(-1))
# # mp = 1.672621925e-27 # in kg
# # me = 9.1093837139e-31 #in kg, but we don't need it 
# me = 1 #in A.U. by definition
# mp = 1.672621925e-27/9.1093837139e-31 * me 
# mu = me*mp/(me+mp)  #TODO convertir en unités atomiques
# print(mu)
# alpha = 1/137.035999177
# n,k = 1,-1 #quantum numbers
# c_original = 1/alpha
# gse_H = mu*c_original**2 / np.sqrt(1 + (Scfs.Z[0]*alpha/(n-np.abs(k) + np.sqrt(k**2 - (Scfs.Z[0]*alpha)**2)))**2)
# # print("Dirac ground state energy of Hydrogen",-13.598433/27.211386245988) #On numerator: Ionisation energy of H in eV (source: NIST), denominator: conversion factor
# print("Dirac ground state energy of Hydrogen",gse_H - c_original**2) #On numerator: Ionisation energy of H in eV (source: Wikipedia, Bohr-Sommerfeld model), denominator: conversion factor
print("Ground state of He in atomic units:", -79.02/27.211386245988)
print(2*(1.43090776) + 2*(-3.37472214) + 2.05163799 - 1.025819)
print(-1.36275874*2)

Ground state of He in atomic units: -2.9039314383202575
-2.86180977
-2.72551748


In [ ]:
#From Luca's orbital4c class
def analytic_1s(light_speed, n, k, Z):
    alpha = 1/light_speed
    gamma = compute_gamma(k,Z,alpha)
    tmp1 = n - np.abs(k) + gamma
    tmp2 = Z * alpha / tmp1
    tmp3 = 1 + tmp2**2
    return light_speed**2 / np.sqrt(tmp3)

def compute_gamma(k,Z,alpha):
    return np.sqrt(k**2 - Z**2 * alpha**2)

c_in = c
n,k = 1,-1 #quantum numbers
# c_in = c_original
gse_H_alt = analytic_1s(c_in, n, k, Scfs.Z[0])
# gse_H_alt = analytic_1s(c_in, n, k, 1)
print(gse_H_alt - c_in**2)

-2.0001065140531864


In [ ]:
Energy_test = -0.50251687
# print((1/Energy_test - 1/(2*137.035999177**2))**(-1))
print((1/Energy_test - 1/(2*10**2))**(-1))

-0.5012574184550684


He 1e-4 c=c

start initMolec
Initmolec
S initial guess =  [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306249392610726+0j)
Expectation value gradk*Nab :  (-0.00012044563871625121+0j)
Expectation value sporb :  1.0530813269418059e-20j
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4306249392610726+0j)
Expectation value gradk*Nab :  (-0.00012044563871625121+0j)
Expectation value sporb :  1.0530813269418059e-20j
Fock [[-0.75587279-9.16960004e-21j]]
Kinetic [[1.43039976-7.60688169e-21j]]
Vnuc [[-3.21193967+0.j]]
Coulomb [[2.05133425+0.j]]
Exchange [[1.02566713+0.j]]
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:170: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu.append(vp.HelmholtzOperator(self.mra, mu, self.prec))  # Initalize the operator
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:650: ComplexWarning: Casting complex values to real discards the imaginary part
  VzFphi = V_z/(self.c**2) * self.Fock[orb, orb] * self.phi_prev[orb][-1]
Power Iter, norm of new guess =  (1.1636000221646157+0j) (1.1636000220968359+0j) (6.777983127744903e-11+0j)
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
init done
scf run
=============Iteration: 0
F= [[-0.75587279-9.16960004e-21j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.234302644185566+1.2164421130315898e-20j)
Expectation value gradk*Nab :  (-0.0001896295500561754+9.27829789425612e-23j)
Expectation value sporb :  (-1.2258840936673134e-09-7.380082494002417e-10j)
/home/qpitto/Tests_KAIN/ZORA/response/src/scfsolv.py:505: ComplexWarning: Casting complex values to real discards the imaginary part
  self.G_mu[orb] = vp.HelmholtzOperator(self.mra, mu, self.prec)
Power Iter, norm of new guess =  (0.9545144369677607+0j) (0.9545144369532508+0j) (1.450994868789787e-11+0j)
orthonorm
S durant Ortho [[0.95451444+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3737119187150915-2.8933942428536424e-14j)
Expectation value gradk*Nab :  (-0.00024151256882254277-4.445931270218589e-14j)
Expectation value sporb :  (2.560565634338832e-10-1.4171918279845763e-10j)
Fock [[-0.92590432+1.18484902e-09j]]
Kinetic [[1.37335916+1.41568302e-10j]]
Vnuc [[-3.30746259-3.63547091e-22j]]
Coulomb [[2.01639822-3.38319294e-22j]]
Exchange [[1.00819911-1.04328071e-09j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.3737119187150915+2.8933942428536424e-14j)
Expectation value gradk*Nab :  (-0.00024151256882254277+4.445931270218589e-14j)
Expectation value sporb :  (3.2221020384703766e-10-1.26383911843977e-10j)
Fock [[-0.92590432-9.16745921e-10j]]
Kinetic [[1.37335916+1.26534792e-10j]]
Vnuc [[-3.30746259+3.63547091e-22j]]
Coulomb [[2.01639822+3.38319294e-22j]]
Exchange [[1.00819911+1.04328071e-09j]]
Orbital: 0    Norm: [0.98572993]    Update: [0.06118879]    Energy:[-0.93510056+7.01888952e-10j]
Scaled energy (4C Dirac energy) [-0.93507728]
=============Iteration: 1
F= [[-0.92590432-9.16745921e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.37361973028909+4.37246138300183e-14j)
Expectation value gradk*Nab :  (-0.00024018197639244103-4.6028255693370674e-14j)
Expectation value sporb :  (2.612019810347377e-10-1.402050945081224e-10j)
Power Iter, norm of new guess =  (0.9839430195371669+0j) (0.983943019534341+0j) (2.825813499590086e-12+0j)
orthonorm
S durant Ortho [[0.98394302+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4277049820484684-2.8428821262221954e-12j)
Expectation value gradk*Nab :  (-0.00025997381502446994+4.010507634517716e-13j)
Expectation value sporb :  (-9.690067467573169e-12+3.3065277030163395e-12j)
Fock [[-0.91819241-5.15848778e-10j]]
Kinetic [[1.4273248-5.7918819e-12j]]
Vnuc [[-3.37092391-5.68685157e-23j]]
Coulomb [[2.05081339-7.61007173e-23j]]
Exchange [[1.02540669+5.10056896e-10j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4277049820484684+2.8428821262221954e-12j)
Expectation value gradk*Nab :  (-0.00025997381502446994-4.010507634517716e-13j)
Expectation value sporb :  (-5.333311036225847e-12+4.279972929752586e-12j)
Fock [[-0.91819241+5.08262278e-10j]]
Kinetic [[1.4273248-1.79461874e-12j]]
Vnuc [[-3.37092391+5.68685157e-23j]]
Coulomb [[2.05081339+7.61007173e-23j]]
Exchange [[1.02540669-5.10056896e-10j]]
Orbital: 0    Norm: [1.00145833]    Update: [0.01703904]    Energy:[-0.92599425+1.183405e-09j]
Scaled energy (4C Dirac energy) [-0.92597142]
=============Iteration: 2
F= [[-0.91819241+5.08262278e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4276055982740383-2.9197647251260125e-12j)
Expectation value gradk*Nab :  (-0.0002584885112870195+4.0382725893439293e-13j)
Expectation value sporb :  (-9.721229913485612e-12+3.2568845232407156e-12j)
Power Iter, norm of new guess =  (0.9989037363648869+0j) (0.998903736364657+0j) (2.2998198792005203e-13+0j)
orthonorm
S durant Ortho [[0.99890374+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4303258964890773+2.8154945952796453e-13j)
Expectation value gradk*Nab :  (-0.00026173652491025983-1.414254251858719e-13j)
Expectation value sporb :  (1.0264391590037613e-11-6.424413012444956e-12j)
Fock [[-0.91815843+6.88959159e-10j]]
Kinetic [[1.42994295+6.56569162e-12j]]
Vnuc [[-3.37359365+7.65142698e-24j]]
Coulomb [[2.05098455-1.15805199e-23j]]
Exchange [[1.02549228-6.82393467e-10j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4303258964890773-2.8154945952796453e-13j)
Expectation value gradk*Nab :  (-0.00026173652491025983+1.414254251858719e-13j)
Expectation value sporb :  (1.3629436514654248e-11-5.539216301476217e-12j)
Fock [[-0.91815843-6.76995529e-10j]]
Kinetic [[1.42994295+5.3979377e-12j]]
Vnuc [[-3.37359365-7.65142698e-24j]]
Coulomb [[2.05098455+1.15805199e-23j]]
Exchange [[1.02549228+6.82393467e-10j]]
Orbital: 0    Norm: [0.99947064]    Update: [0.00121474]    Energy:[-0.91828927-5.15871215e-10j]
Scaled energy (4C Dirac energy) [-0.91826682]
=============Iteration: 3
F= [[-0.91815843-6.76995529e-10j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4302256079736853+3.0267657892122255e-13j)
Expectation value gradk*Nab :  (-0.0002602359558034085-1.4238810454377692e-13j)
Expectation value sporb :  (1.0256621032247844e-11-6.4393809861854615e-12j)
Power Iter, norm of new guess =  (0.9995344453686966+0j) (0.9995344453686966+0j) 0j
orthonorm
S durant Ortho [[0.99953445+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4311335158716172-5.171269015102367e-13j)
Expectation value gradk*Nab :  (-0.0002620226325307716+2.3568380886061338e-14j)
Expectation value sporb :  (9.14147467057904e-14-2.1533349830497915e-14j)
Fock [[-0.91802105+4.87722836e-11j]]
Kinetic [[1.43075016-4.76653029e-13j]]
Vnuc [[-3.37453882+8.03189155e-24j]]
Coulomb [[2.05153523+5.12464239e-24j]]
Exchange [[1.02576762-4.92489366e-11j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4311335158716172+5.171269015102367e-13j)
Expectation value gradk*Nab :  (-0.0002620226325307716-2.3568380886061338e-14j)
Expectation value sporb :  (1.6024447498483704e-14-3.6287965135504856e-14j)
Fock [[-0.91802105-4.87144622e-11j]]
Kinetic [[1.43075016+5.34474344e-13j]]
Vnuc [[-3.37453882-8.03189155e-24j]]
Coulomb [[2.05153523-5.12464239e-24j]]
Exchange [[1.02576762+4.92489366e-11j]]
Orbital: 0    Norm: [0.99999817]    Update: [0.00022811]    Energy:[-0.91825616+6.88993557e-10j]
Scaled energy (4C Dirac energy) [-0.91823371]
=============Iteration: 4
F= [[-0.91802105-4.87144622e-11j]]
S= [[1.+0.j]]
orthonorm
S durant Ortho [[1.+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4310330421922295-5.200046958476922e-13j)
Expectation value gradk*Nab :  (-0.00026051911597895297+2.371703873864434e-14j)
Expectation value sporb :  (9.138295158245189e-14-2.1517369574866386e-14j)
Power Iter, norm of new guess =  (0.9997722427554468+0j) (0.9997722427554468+0j) 0j
orthonorm
S durant Ortho [[0.99977224+0.j]]
Post-Ortho S= [[1.+0.j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4312912068068082-1.472237995902711e-14j)
Expectation value gradk*Nab :  (-0.0002620851900043798+2.0361026678828948e-14j)
Expectation value sporb :  (1.5319242460279975e-14-1.4292718075487608e-13j)
Fock [[-0.91799539-1.71548646e-11j]]
Kinetic [[1.43090776+1.45760698e-13j]]
Vnuc [[-3.37472214-4.91138482e-25j]]
Coulomb [[2.05163799-6.62390726e-25j]]
Exchange [[1.025819+1.73006253e-11j]]
compFock main loop 0
Expectation value (k-1/2)*nabla^2 :  (1.4312912068068082+1.472237995902711e-14j)
Expectation value gradk*Nab :  (-0.0002620851900043798-2.0361026678828948e-14j)
Expectation value sporb :  (3.7664955590040124e-13-5.28110030754466e-14j)
Fock [[-0.91799539+1.73506028e-11j]]
Kinetic [[1.43090776+4.99774856e-14j]]
Vnuc [[-3.37472214+4.91138482e-25j]]
Coulomb [[2.05163799+6.62390726e-25j]]
Exchange [[1.025819-1.73006253e-11j]]
Orbital: 0    Norm: [0.99999996]    Update: [4.20171306e-05]    Energy:[-0.91811896+4.87696427e-11j]
Scaled energy (4C Dirac energy) [-0.91809652]